src/services/AIService.js

```diff
@@ -0,0 +1,20 @@
+import { prepareApiRequest } from 'src/boot/axios';
+
+/**
+ * Generate diagrams using Leto AI proxy.
+ * It posts a request to the proxy and returns a json response composed of an
+ * array with each item being a diagram (a name and a content).
+ * @param {string} plugin - name of the plugin
+ * @param {string} description - description of the diagram
+ * @returns {Promise<object>} Promise with the diagram information on success
+ * otherwise an error.
+ */
+export async function generateDiagram(plugin, description) {
```

Comment:
A default timeout is configured to 15s, configure it for 15min or another increased time.